src/libraries/Common/src/System/Security/Cryptography/SP800108HmacCounterKdfImplementationCng.cs

```diff
@@ -0,0 +1,213 @@
+            {
+                DeriveBytes(labelData.Utf8Bytes, contextData.Utf8Bytes, destination);
+            }
+        }
+
+        internal static void DeriveBytesOneShot(
+            ReadOnlySpan<byte> key,
+            HashAlgorithmName hashAlgorithm,
+            ReadOnlySpan<byte> label,
+            ReadOnlySpan<byte> context,
+            Span<byte> destination)
+        {
+            Debug.Assert(destination.Length <= 0x1FFFFFFF);
+
+            using (SP800108HmacCounterKdfImplementationCng kdf = new SP800108HmacCounterKdfImplementationCng(key, hashAlgorithm))
+            {
+                kdf.DeriveBytes(label, context, destination);
+            }
+        }
+
+        internal static void DeriveBytesOneShot(
+            ReadOnlySpan<byte> key,
+            HashAlgorithmName hashAlgorithm,
+            ReadOnlySpan<char> label,
+            ReadOnlySpan<char> context,
+            Span<byte> destination)
+        {
+            if (destination.Length == 0)
+            {
+                return;
+            }
+
+            using (Utf8DataEncoding labelData = new Utf8DataEncoding(label, stackalloc byte[CharToBytesStackBufferSize]))
+            using (Utf8DataEncoding contextData = new Utf8DataEncoding(context, stackalloc byte[CharToBytesStackBufferSize]))
+            {
+                DeriveBytesOneShot(key, hashAlgorithm, labelData.Utf8Bytes, contextData.Utf8Bytes, destination);
+            }
+        }
+
+        private static unsafe SafeBCryptKeyHandle CreateSymmetricKey(byte* symmetricKey, int symmetricKeyLength)
+        {
+            NTSTATUS generateKeyStatus;
+            SafeBCryptKeyHandle keyHandle;
+
+            if (s_sp800108CtrHmacAlgorithmHandle is not null)
+            {
+                generateKeyStatus = Interop.BCrypt.BCryptGenerateSymmetricKey(
+                    s_sp800108CtrHmacAlgorithmHandle,
+                    out keyHandle,
+                    pbKeyObject: IntPtr.Zero,
+                    cbKeyObject: 0,
+                    symmetricKey,
+                    symmetricKeyLength,
+                    dwFlags: 0);
+            }
+            else
+            {
+                generateKeyStatus = Interop.BCrypt.BCryptGenerateSymmetricKey(
+                    BCRYPT_SP800108_CTR_HMAC_ALG_HANDLE,
+                    out keyHandle,
+                    pbKeyObject: IntPtr.Zero,
+                    cbKeyObject: 0,
+                    symmetricKey,
+                    symmetricKeyLength,
+                    dwFlags: 0);
+            }
+
+            if (generateKeyStatus != NTSTATUS.STATUS_SUCCESS)
+            {
+                keyHandle.Dispose();
+                throw Interop.BCrypt.CreateCryptographicException(generateKeyStatus);
+            }
+
+            Debug.Assert(!keyHandle.IsInvalid);
+
+            return keyHandle;
+        }
+
+        /// <summary>
+        /// Returns null if the platform ins Windows 10+ and psuedo handles should be used.
```

Comment:
It probably only wants to be a `//` comment instead of xmldoc... but either way "ins" isn't English :smirk:

```suggestion
        /// Returns null if the platform is Windows 10+ and psuedo handles should be used.
```